sig-security-assessments/cluster-api/self-assessment.md

```diff
@@ -0,0 +1,1099 @@
+  control on Cluster API controller namespaces to prevent unauthorised access by
+  anyone other than cloud admin.
+  - Status: Implemented for K8s RBAC, developer guidance to be written for
+    providers
+* Enable auditing on the management cluster as well as on the cloud provider
+  * Status: [End user guidance to be written](#cluster-api-end-user-guide)
+
+#### STRIDE-SPOOF-2 Spoofing of cloud provider services
+
+Cloud provider services like Secrets Manager store sensitive data like cloudinit
+configuration files
+(shortlived). Other services like EC2 and ELB API have the ability to sniff
+traffic destined to the API server of the workload cluster. Successful spoofing
+of a cloud service could result in information disclosure and tampering of in
+flight requests relevant to cluster and machine management.
+
+#### STRIDE-SPOOF-2 Recommended Mitigations
+
+* Verify server side identity to confirm that AWS service is indeed what it
+  advertises to be.
+  * Status: Implemented (vSphere) / Out-of-scope (AWS/Azure)
+  * AWS maintainers note: We rely on
+    the [AWS Shared Responsibility model](#AWS-Shared-Responsibility-model).
+    There is realistically little we can do in the event of a major breach.
+    Validation of server endpoints is left to the standard SDK.
+
+#### STRIDE-SPOOF-3 Spoofing of image-builder maintainer
+
+Description: Machine image for cluster is pre-baked and built by cluster API
+maintainers. This image could be tampered with to include a cryptominer if the
+accounts of maintainers that have privileges and ownership over the code and
+machine image built for Nodes in a cluster.
+
+#### STRIDE-SPOOF-3 Recommended Mitigations
+
+* Implement 2FA for all maintainer accounts on Github. Apply the second pair of
+  eyes principle when performing privileged actions such as image building or
+  updates to the contents of the machine images.
+  * Status: Implemented
+  * AWS maintainers note: Public community consumed images are currently
+    maintained by VMware staff. Access to this environment is 2FA protected and
+    managed by an SRE team. In order to give more confidence to the community,
+    it is planned to transfer image building to CNCF owned accounts.
+
+#### STRIDE-SPOOF-4 Spoofing of a control plane node in a workload cluster
+
+When a worker node is created, a user could somehow interfere with the boot,
+e.g. breaking networking, assigning incorrect security groups. This would
+prevent the encrypted cloud-init data from being deleted. If the user could gain
+access to the node, they could use the IMDS provided credentials to read the
+bootstrap token from Secrets Manager and attempt to register AS a control plane
+node, since validation is only done client- side by kubeadm. This can allow the
+attacker to gain access to secrets they should not be able to.
+
+#### STRIDE-SPOOF-4 Recommended Mitigations
+
+* Replace kubeadm node join mechanism with a host attestation mechanism.
+  * Status: Planned
```

Comment:
Is there something that we can link? (same below)

Reply:
Linking the CAEP definitely makes sense. I found this issue: https://github.com/kubernetes-sigs/cluster-api/issues/3762 which I think corresponds to the proposal, but they are not linked via GitHub for some reason.